src/library/Uno.Material/Converters/FromTextBoxEmptyStringToValueConverter.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+#if WinUI
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Data;
+#else
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Data;
+#endif
+
+namespace Uno.Material
+{
+	public class FromTextBoxEmptyStringToValueConverter : IValueConverter
+	{
+		public object NoHeaderAndNoPlaceholderValue { get; set; }
+
+		public object HeaderOnlyValue { get; set; }
+
+		public object HeaderOnlyWithTextValue { get; set; }
+
+		public object PlaceholderOnlyValue { get; set; }
+
+		public object HeaderAndPlaceholderValue { get; set; }
+
+		public object Convert(object value, Type targetType, object parameter, string language)
+		{
+			var textBox = value as TextBox;
+
+			if (textBox is not null)
+			{
+				var noHeader = !(textBox.Header is string str1) || string.IsNullOrEmpty(str1);
+				var noPlaceholder = !(textBox.PlaceholderText is string str2) || string.IsNullOrEmpty(str2);
+				var noText = string.IsNullOrEmpty(textBox.Text);
```

Comment:
If i remember correctly, IValueConverter::Convert is only ever called if the binding target results in a different value, which in the case here would be the TextBox itself. The problem would be if you adjust the Header/PlaceholderText/Text properties after the initial binding, `::Convert` will not be called as the value would remain the same.

Reply:
looks like this is fine when used in setter value converter